### PR TITLE
Add Room database implementation

### PR DESCRIPTION
**Pull request by nicholas-cook**
- Adds support for Room
- Creates a database to hold Categories and Phrases
- Add DAOs and a repository class to handle querying the database
- Hooks up the `PresetsViewModel` to use this new Room data source
- Updates fragments to use model objects instead of strings

### CODE REVIEW

File: app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import android.content.Context
+import com.willowtree.vocable.room.Category
+import com.willowtree.vocable.room.Phrase
+import com.willowtree.vocable.room.VocableDatabase
+import org.koin.core.KoinComponent
+import org.koin.core.get
+
+class PresetsRepository(context: Context) : KoinComponent {
+
+    private val database = VocableDatabase.getVocableDatabase(context)
```
Comment (dnylng) [inline app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt]: Why don't you set up a Koin module for your DB and inject it into your repository class?

File: app/src/main/java/com/willowtree/vocable/presets/PresetsViewModel.kt [hunk 2]
```diff
@@ -23,7 +25,7 @@ class PresetsViewModel : ViewModel(), KoinComponent {
         const val CATEGORY_MY_SAYINGS = "My Sayings"
     }
 
-    private val sharedPrefs: VocableSharedPreferences by inject()
+    private val presetsRepository: PresetsRepository by inject()
```
Comment (dnylng) [inline app/src/main/java/com/willowtree/vocable/presets/PresetsViewModel.kt]: You should definitely check out using viewModel injection that Koin is offering.

File: app/src/main/java/com/willowtree/vocable/room/PhraseDao.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertPhrase(phrase: Phrase)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertPhrases(vararg phrases: Phrase)
+
+    @Delete
+    suspend fun deletePhrase(phrase: Phrase)
+
+    @Query("SELECT * FROM Phrase WHERE is_user_generated == 1")
```
Comment (dnylng) [inline app/src/main/java/com/willowtree/vocable/room/PhraseDao.kt]: Not sure if this breaks anything but is it supposed to be `WHERE is_user_generated == 1` or `WHERE is_user_generated = 1`?

Comment (dnylng) [reply app/src/main/java/com/willowtree/vocable/room/PhraseDao.kt]: Cuz this is based on mySQL comparisons and not kotlin/java lang right?

File: app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt (hunk 1 above)
Comment (nicholas-cook) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsRepository.kt]: 👍 We're going to look into this once we get the saving presets functionality working.

File: app/src/main/java/com/willowtree/vocable/room/PhraseDao.kt (hunk 3 above)
Comment (nicholas-cook) [reply app/src/main/java/com/willowtree/vocable/room/PhraseDao.kt]: It looks like Room accepts both `=` and `==` here

**Pull request merged**